src/ai/backend/common/types.py

```diff
@@ -1894,3 +1895,8 @@ class SessionExecutionStatus(enum.StrEnum):
     FINISHED = "finished"
     CANCELED = "canceled"
     TIMEOUT = "timeout"
+
+
+class StreamReader(Protocol):
+    def read(self) -> AsyncIterator[bytes]: ...
+    def content_type(self) -> Optional[str]: ...
```

Comment:
For this current purpose, it is correct to inherit ABC. Protocol is not intended for inheritance, but is for situations where there is duck typing in multiple classes, so the usage is different.